scalableminds/webknossos
# Remember the forcefully disabled tool and reenable it if possible

This PR tackles the problem described in #6293. When shortly disabling the segmentation layer via shortcut `3` to only see the color data the volume tool was deselected. The user then had to manually reselect the previous volume tool. To speed up annotating, the forcefully disabled tool is now remembered and reselected again if it is no longer disabled and no other tool has been selected after this tool was disabled.

### URL of deployed dev instance (used for testing):
- https://rememberactivetool.webknossos.xyz

### Steps to test:
- Open an annotation with a segmentation layer.
- Select the brush tool and hit `3` on the keyboard to disable the segmentation layer.
- The move tool should now be selected as the brush is disabled.
- Hit `3` again and the brush tool should now be automatically selected.
- Do the same for the other volume tools. This should also work.
- Now select the brush tool, hit `3`, and then select the skeleton tool. Hitting `3` again should not select the brush tool.
-  At last, select the brush tool, hit `3`, select the skeleton tool, and then select the move tool again. Hitting `3` again should not select the brush tool as the user intentionally changed the tools in between disabling and enabling the segmentation layer.

### Issues:
- fixes #6293

------
(Please delete unneeded items, merge only when none are left open)
